xdsl/dialects/func.py

```diff
@@ -69,6 +69,53 @@ def from_region(name: str, input_types: List[Attribute],
         op = FuncOp.build(attributes=attributes, regions=[region])
         return op
 
+    def replace_argument_type(self, arg: int | BlockArgument,
+                              new_type: Attribute):
+        """
+        Replaces a function argument with a new one
```

Comment:
Please add documentation that this is refering to the arg type of the entry block.

Reply:
Also, should this API really be two methods, one of which makes the op not verify anymore? I think inlining the other function into this one would give a cleaner API.

Reply:
Ah nvm, the second one does not invalidate the op. The two function part is fine, please still add some docu :)

Reply:
Added docu :D